File: Day 1/javaScript/Level 1/LevelOne.js
function arraySplitter(array){

    let evenNumbers = [];
    let oddNumbers = [];

    for(let number of array){

        if(number % 2 === 0){
            evenNumbers.push(number);
        }

        else{
            oddNumbers.push(number);
        }
    }

    return [evenNumbers, oddNumbers];
}


function palindromicArray(array){

    for(let count = 0; count < array.length / 2; count++){

        if(array[count] !== array[array.length - 1 - count]){
            return false;
        }
    }

    return true;
}


function getPerfectSquare(array){

    let perfectSquares = [];

    for(let number of array){

        let root = Math.floor(Math.sqrt(number));

        if(root * root === number){
            perfectSquares.push(number);
        }
    }

    return perfectSquares;
}


function replaceNonPerfectSquare(array){

    let result = [];

    for(let number of array){

        let root = Math.floor(Math.sqrt(number));

        if(root * root === number){
            result.push(number);
        }

        else{
            result.push(-1);
        }
    }

    return result;
}


module.exports = {
    arraySplitter,
    palindromicArray,
    getPerfectSquare,
    replaceNonPerfectSquare
};

File: Day 1/python/level 1/LevelOne.py
def array_splitter(array):

    even_numbers = []
    odd_numbers = []

    for number in array:

        if number % 2 == 0:
            even_numbers.append(number)

        else:
            odd_numbers.append(number)

    return [even_numbers, odd_numbers]


def palindromic_array(array):

    for count in range(len(array) // 2):

        if array[count] != array[len(array) - 1 - count]:
            return False

    return True


def get_perfect_square(array):

    perfect_squares = []

    for number in array:

        root = int(number ** 0.5)

        if root * root == number:
            perfect_squares.append(number)

    return perfect_squares


def replace_non_perfect_square(array):

    result = []

    for number in array:

        root = int(number ** 0.5)

        if root * root == number:
            result.append(number)

        else:
            result.append(-1)

    return result